projects/python-contract-examples/smart_contracts/arc4_types/contract.py

```diff
@@ -0,0 +1,305 @@
+        """
+        return arc4.Byte(a.native + 1)
+
+    # example: ARC4_BYTES
+
+    # example: ARC4_ADDRESS
+    @abimethod()
+    def arc4_address_properties(self, address: arc4.Address) -> UInt64:
+        underlying_bytes = (  # noqa: F841
+            address.bytes
+        )  # This will return the underlying bytes of the address.
+
+        account = (
+            address.native
+        )  # This will return the account type of the given address.
+
+        bal = account.balance  # returns the balance of the account
+        total_asset = (  # noqa: F841
+            account.total_assets
+        )  # returns the total assets held in the account
+
+        return bal
+
+    @abimethod()
+    def arc4_address_return(self, address: arc4.Address) -> arc4.Address:
+
+        account = (
+            address.native
+        )  # This will return the account type of the given address.
+
+        """
+        You can't return an Account type because it is a reference type.
+        Convert the Account type to arc4.Address type and return it.
+        """
+        converted_address = arc4.Address(account)
+
+        assert converted_address == address
+
+        return converted_address
+
+    # example: ARC4_ADDRESS
+
+
+# example: ARC4_STATIC_ARRAY
+AliasedStaticArray: t.TypeAlias = arc4.StaticArray[arc4.UInt8, t.Literal[1]]
+
+
+class Arc4StaticArray(ARC4Contract):
+
+    @abimethod()
+    def arc4_static_array(self) -> None:
+        """
+        You can create a static array directly from the contract.
+        """
+        static_uint32_array = arc4.StaticArray(
+            arc4.UInt32(1), arc4.UInt32(10), arc4.UInt32(255), arc4.UInt32(128)
+        )
+
+        total = UInt64(0)
+        for uint32_item in static_uint32_array:
+            total += uint32_item.native
+
+        assert total == 1 + 10 + 255 + 128
+
+        """
+        You can also create a static array using a type alias.
+        """
+        aliased_static = AliasedStaticArray(arc4.UInt8(101))
+
+        index = UInt64(0)
+
+        assert (aliased_static[0].native + aliased_static[index].native) == 202
+
+        aliased_static[0] = arc4.UInt8(202)
+
+        assert aliased_static[0].native == 202
+
+        """
+        You can't add or pop values from a static array because it has a fixed size.
+        so this won't compile:
+        aliased_static.pop()
+        """
+
+
+# example: ARC4_STATIC_ARRAY
+
+# example: ARC4_DYNAMIC_ARRAY
+goodbye: t.TypeAlias = arc4.DynamicArray[arc4.String]
+
+
+class Arc4DynamicArray(ARC4Contract):
+
+    @abimethod
+    def goodbye(self, name: arc4.String) -> goodbye:
+        bye = goodbye(arc4.String("Good bye "), name)
+
+        return bye
+
+    @abimethod()
+    def hello(self, name: arc4.String) -> String:
+        """
+        Dynamic Arrays have variable size and capacity.
+        They are similar to native Python lists because they can also append, extend, and pop.
```

Comment:
```suggestion
        They are similar to native Python lists because they can also append, extend, and pop,
        but differ in being type-restricted and having a fixed maximum size of 4096 bytes.
```